pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -2,13 +2,222 @@ package awsloadbalancercontroller
+				MatchLabels: map[string]string{
+					appLabelName:    appName,
+					appInstanceName: controller.Name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						appLabelName:    appName,
+						appInstanceName: controller.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+
+					Containers: []corev1.Container{
+						{
+							Name:  "controller",
+							Image: image,
+							Args:  desiredContainerArgs(controller, clusterName, vpcID),
+						},
+					},
+					ServiceAccountName: sa.Name,
+				},
+			},
+		},
+	}
+	if controller.Spec.Config != nil && controller.Spec.Config.Replicas != 0 {
+		d.Spec.Replicas = pointer.Int32(controller.Spec.Config.Replicas)
+	}
+	return d
+}
+
+func desiredContainerArgs(controller *albo.AWSLoadBalancerController, clusterName, vpcID string) []string {
+	var args []string
+	args = append(args, fmt.Sprintf("--aws-vpc-id=%s", vpcID))
+	args = append(args, fmt.Sprintf("--cluster-name=%s", clusterName))
+
+	// if additional keys are present then sort them and append it to the arguments
+	if controller.Spec.AdditionalResourceTags != nil {
+		var tags []string
+		for k, v := range controller.Spec.AdditionalResourceTags {
+			tags = append(tags, fmt.Sprintf("%s=%s", k, v))
+		}
+		sort.Strings(tags)
+		args = append(args, fmt.Sprintf(`--default-tags=%s`, strings.Join(tags, ",")))
+	}
+	args = append(args, "--disable-ingress-class-annotation")
+	args = append(args, "--disable-ingress-group-name-annotation")
+	if controller.Spec.Config != nil && controller.Spec.Config.Replicas > 1 {
+		args = append(args, "--enable-leader-election")
+	}
+	enabledAddons := make(map[albo.AWSAddon]struct{})
+	for _, a := range controller.Spec.EnabledAddons {
+		enabledAddons[a] = struct{}{}
+	}
+	if _, ok := enabledAddons[albo.AWSAddonShield]; ok {
+		args = append(args, "--enable-shield=true")
+	} else {
+		args = append(args, "--enable-shield=false")
+	}
+	if _, ok := enabledAddons[albo.AWSAddonWAFv1]; ok {
+		args = append(args, "--enable-waf=true")
+	} else {
+		args = append(args, "--enable-waf=false")
+	}
+	if _, ok := enabledAddons[albo.AWSAddonWAFv2]; ok {
+		args = append(args, "--enable-wafv2=true")
+	} else {
+		args = append(args, "--enable-wafv2=false")
+	}
+	args = append(args, fmt.Sprintf("--ingress-class=%s", controller.Spec.IngressClass))
+	sort.Strings(args)
+	return args
+}
+
+func (r *AWSLoadBalancerControllerReconciler) currentControllerDeployment(ctx context.Context, name string, namespace string) (bool, *appsv1.Deployment, error) {
+	var deployment appsv1.Deployment
+	err := r.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, &deployment)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+	return true, &deployment, nil
+}
+
+func (r *AWSLoadBalancerControllerReconciler) createDeployment(ctx context.Context, deployment *appsv1.Deployment) error {
```

Comment:
Not a big deal, but just noticed a small discrepancy in the naming of the methods: `ensureControllerDeployment` and `currentControllerDeployment` but `createDeployment` and `updateDeployment`.